### PR TITLE
Add non-tech docs for chat assignment

### PR DESCRIPTION
**Pull request by psiyan**
https://hello-haptik.atlassian.net/browse/AGCT-259

### CODE REVIEW

File: docs/agent-chat/chat-assignment.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+## What does this document discuss?
+
+Now let's look at typical use cases of the chat assignment system:
+
+
+| Use case                                               | Analogy                                                      | Changed? | Discussed here? |
+| ------------------------------------------------------ | ------------------------------------------------------------ | -------- | --------------- |
+| Assigning users' chats to a team                       | Which brigade is going to attend the service?                | No       | No              |
+| Selecting which user needs to be addressed next        | Of the waiting guests, who will be served next?              | No       | No              |
```
Comment (deepwired) [inline docs/agent-chat/chat-assignment.md:30]: Assigning users' chats to a team  and Selecting which user needs to be addressed next , under discussed here, lets use words like "out of scope of chat assignment" or something.

File: docs/agent-chat/chat-assignment.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+|       |         |        |            |
+|       |         |        |            |
+
+
+
+
+### Deciding agent priority
+
+Given the complexity of the assignment algorithms,
+we're unable to share a deterministic answer to how exactly chats are assigned
```
Comment (deepwired) [inline docs/agent-chat/chat-assignment.md:69]: we should remove this line.

File: docs/agent-chat/chat-assignment.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    it's possible that her slot 1 and 2 get allocated first amongst the entire team.
+    
+    While her slot 3 is allocated after all other slots of her colleagues have been exhausted. 
+ 2. **Agent priority is directly proportional to:**
+    - **number of free slots of the agent**
+    - **amount of time the agent has been waiting for a chat to be assigned to them**
+
+  > NOTE: Agent priority is not calculated at the time a chat arrives in the system,
+    rather we try to pre-determine who is most likely to be free in the near future. 
+    Hence, it is possible to have multiple chats assigned in succession to an agent at an instance,
```
Comment (deepwired) [inline docs/agent-chat/chat-assignment.md:84]: i thought we removed this?

File: docs/agent-chat/chat-assignment.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    if they had not taken chats in a long time.
+    This is done to avoid delaying chat assignment in real-time due to time spent in computing the current priority.
+
+
+#### An example
+
+Let's assume a typical day, where the agents came online in an alphabetical order of their names.
+
+
+| Time | Agent | Concurrency |
```
Comment (deepwired) [inline docs/agent-chat/chat-assignment.md:94]: we could show agents in team with status information, to better convey the message.

File: docs/agent-chat/chat-assignment.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+|       |         |        |            |
+
+
+
+
+### Deciding agent priority
+
+Given the complexity of the assignment algorithms,
+we're unable to share a deterministic answer to how exactly chats are assigned 
+
```
Comment (nikunj1705) [inline docs/agent-chat/chat-assignment.md:70]: We cannot say this in a client facing document. If we don't know the answer, who does?

File: docs/agent-chat/chat-assignment.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+### The setting
+
+In our restaurant, there are only two kind of people involved in the operation:
+ - Chefs who are responsible for receiving the customers and taking their order, preparing the meal, and serving it.
+ - Guests who come in and sit down at the table in the restaurant and wait to be attended to.
+
+Obviously, due to COVID-19, the guest's don't actually eat at the restaurant.
+They just sit, physically distancing from each other, waiting for their food to be prepared.
+Once a chef returns with their meal, they pick it up and leave the restaurant.
+
```
Comment (nikunj1705) [inline docs/agent-chat/chat-assignment.md:21]: Do we need the restaurant analogy? Let's share details about the old algorithm and the new algorithm. What are the pros and cons for each. One does assignment in a certain way and the other does X.

File: docs/agent-chat/chat-assignment.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+ 2. **Agent priority is directly proportional to:**
+    - **number of free slots of the agent**
+    - **amount of time the agent has been waiting for a chat to be assigned to them**
+
+  > NOTE: Agent priority is not calculated at the time a chat arrives in the system,
+    rather we try to pre-determine who is most likely to be free in the near future. 
+    Hence, it is possible to have multiple chats assigned in succession to an agent at an instance,
+    if they had not taken chats in a long time.
+    This is done to avoid delaying chat assignment in real-time due to time spent in computing the current priority.
+
```
Comment (nikunj1705) [inline docs/agent-chat/chat-assignment.md:87]: Line 71 to 86 is bang on. This is needed.

File: docs/agent-chat/chat-assignment.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+| Brinda's Slot 2 |
+| Carly's Slot 2  |
+| Brinda's Slot 3 |
+
+## FAQs
+
+ - What are the factors do you consider to assign chats?
+
+   This question is too vague 
+
```
Comment (nikunj1705) [inline docs/agent-chat/chat-assignment.md:142]: You cannot put that in an FAQ. The question is too vague.

File: docs/agent-chat/chat-assignment.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+| Slot priorities |
+| --------------- |
+| Alice's Slot 1  |
+| Brinda's Slot 1 |
+| Alice's Slot 2  |
+| Brinda's Slot 2 |
+| Brinda's Slot 3 |
+
+------
+
```
Comment (nikunj1705) [inline docs/agent-chat/chat-assignment.md:120]: Can we give a simpler example. Like the one given in Chat assignment PRD. Just that agent 1 has x available slots and agent 2 has y slots. The chat will go to agent 1. Adding tables here might be confusing for clients.

**Pull request closed without merging**